APTITUDE/APTITUDETEST.Data/Data/UserRepository.cs

```diff
@@ -351,35 +350,62 @@ public async Task<JsonResult> DeleteUsers(List<int> userIds)
         #endregion
 
         #region ImportUsers
-        public async Task<JsonResult> ImportUsers(IFormFile file)
+        public async Task<JsonResult> ImportUsers(ImportUserVM importUsers)
         {
             try
             {
-                using (var reader = new StreamReader(file.OpenReadStream()))
+                using (var reader = new StreamReader(importUsers.file.OpenReadStream()))
                 using (var csv = new CsvReader(reader, CultureInfo.InvariantCulture))
                 {
                     var records = csv.GetRecords<UserImportVM>().ToList();
+                    if (records.Count <= 0)
+                    {
+                        return new JsonResult(new ApiResponse<int>
+                        {
+                            Message = string.Format(ResponseMessages.BadRequest),
+                            Result = false,
+                            StatusCode = ResponseStatusCode.BadRequest
+                        });
+                    }
+                    foreach (var record in records)
+                    {
+                        record.groupid = importUsers.GroupId;
+                        record.collegeid = importUsers.CollegeId;
+                    }
                     if (records != null)
                     {
+                        int result;
                         using (var connection = _dapperContext.CreateConnection())
                         {
-                            var procedure = "import_users";
-                            var parameters = new DynamicParameters(
-                            new
-                            {
-                                p_import_user_data = records.ToArray()
-                            });
+                            var procedure = "import_users_new";
+                            var parameters = new DynamicParameters();
+                            parameters.Add("p_import_user_data", records, DbType.Object, ParameterDirection.Input);
+                            parameters.Add("candidates_added_count", ParameterDirection.Output);
+
+                            // Execute the stored procedure
+
+                            result = connection.Query<int>(procedure, parameters, commandType: CommandType.StoredProcedure).FirstOrDefault();
 
```

Comment:
As per our discussion, Update this stored procedure - pass groupId and collegeId